config/dcp.go

```diff
@@ -43,6 +43,13 @@ const (
 	KubernetesLeaderElectorRetryPeriodConfig        = "retryPeriod"
 )
 
+type DcpMode string
+
+const (
+	Infinite DcpMode = "infinite"
```

Comment:
Adding enums with constant prefix is more appropriate in Go's package model.
For example, currently, you need to reference like `config.Infinite` or `config.Finite` but this is not say exactly what it is.
Instead,
use constant prefix like 
```
DcpModeInfinite DcpMode = "infinite"
DcpModeFinite   DcpMode = "finite"
```

refer it like `config.DcpModeInfinite` or `config.DcpModeFinite` is more appropriate.

You can also find lots of usage like this in Go's and it's popular repositories.